### PR TITLE
Radiation funtimes

### PR DESCRIPTION
**Pull request by ghost**
Previously the hardcap would mean that, should you be near a radsource, you'll be stuck in faint city until death or limbo as it'd keep setting your radiation to 100, and if you were still near the radiation emitter it would tick that right back up. It also meant that radiation itself would only really stay in your system for a hundred ticks or so depending on exposure, making chems to treat radiation exposure such as hyronalin or arithrazine pointless.

No longer will radstorms be a collapse-spam of braindeads, if you stand in the fire, be prepared to stew.

- [x] Merge this shit when ticked, otherwise it's still a WIP

TODO:

- [x] Make radiation exposure more horrifying/interesting
- [x] Perhaps add more ghetto alternatives for treating radiation exposure
///- [ ] Get shouted at because I annoyed somebody with this
- [x] Get shouted at because I fucked something up

Potential radiation ailments

6 'gates'
		100
			Vomiting - Early on symptom, if rad_tick and radiation is above a certain level 
			Nosebeeds - Early onset of the blood vessels breaking down
		200
			Internal hemorrhaging - Lower rad tick has lower probability, but this ramps up as it goes along
		300
			Organ Damage - A little later on, again lower rad tick has lower prob
		400
			Minor limb mutation - We're talking a decently large exposure causing this, somewhat rare
			Blindness - About the same chance as limb mutation
			Clone damage - To emulate DNA breakdown
		500
			Necrosis - After Limb mutation
			Glowing slightly green - Extremely rare, practically terminal, glowing increases with radiation level
		600
			Ghoulification if extremely lucky - We're talking 1 in 1000

:cl:
* rscadd: Radiation is now no longer stunlock city
* rscadd: Radiation is now a trip to hell instead
* rscadd: Ghouls added
* rscadd: Greyshirt vodka added, made from crushed cockroach, crushed potato, and water

### CODE REVIEW

**Comment by AlphaBuffalo** [comment]
I can check that last tick for you, YOU BROKE MY SANITY BECAUSE I THOUGHT NO ONE WOULD EVER DO THIS!

**Comment by amikawaii** [comment]
Radiologically trained individual (fuck off skowron).  
Add an organ damage mechanic.
The end game of high rad levels would result in significant internal bleeding as a result of the cells that make up capilaries and other vessels die and tear apart.

Vomiting is a common symtpom of medium dose levels, but with most of these it isnt going to be an immediate thing, the guys on K19 still made it home, mostly.  

I think organ damage is the way to go.  
You could consider adding radioactive particulate.  
Make it so that the SME can put airborne paritcles that add whatever element (radium or something) to the body if you arent on internals, or add potassium iodine so airborne radioactivity affects you less.  Sorry for rambling been drinking

**Comment by ghost** [comment]
That ain't rambling, that's gold.

So, that's the IRL stuff covered, how about some more sci fi elements?

Vomit glows green, perhaps?

**Comment by Orangerush** [comment]
ACID VOMIT???

**Comment by AlphaBuffalo** [comment]
I assume random mutations are going to stay, that's pretty sci fi with our genes

**Comment by ghost** [comment]
random mutations is via radstorms, it's snowflaked like that.

**Comment by Orangerush** [comment]
I wanna see radiation really fuck up people instead of WOOPS YOU GOT SOME TOXIN DAMAGE AND NOW YOU BORK BORK BORK

**Comment by ghost** [comment]
Then give ideas!

Silly ones, outlandish ones, simple ones, ones that make you laugh, ones that make you cry from fear.

Give me those sweet ideas to make into reality.

**Comment by SkowronX** [comment]
@amikawaii 
literally stole all my """ideas"""

**Comment by AI-Is-Rogue** [comment]
Call advanced_mutate on carbon mobs? Like phazon does.

**Comment by Orangerush** [comment]
Instead of saying you're glowing actually have you glow a bright green like feral ghouls. Acid vomit or janitors to have to clean up. Physical mutations like arms changing into something fucked up, have limb flesh fall off randomly

**Comment by unid15** [comment]
I was about to suggest that, after extreme irradiation some limbs should change into skeleton ones, as flesh rots off of them

**Comment by AlphaBuffalo** [comment]
What about the chance to grow a useless 3rd arm?

**Comment by Megamaw** [comment]
honestly, since skeletons are just skeletons, while you're doing this you could probably give skeletons rad resist so they have an actual upside

we shouldn't aim to make radiation damage 'fun and hilarious' like speaking in comic sans or other ridiculous genetic effects. it's not commonly weaponized by antags, but being rad damaged should be an absolute nightmare.

and i mean that not necessarily from a gameplay sense, but from a 'i'm gonna fucking puke' kind of sense. but how far can we go without making spriting a living nightmare? uhhh, vestigial gross things coming out of the head? skin sloughing off until they bleed? blindness from eyelids being forced shut?


why do i think it should be absolutely nightmarish? well, i mean, after this is made a thing, how will we decide what becomes a 'power/defect' and what becomes a 'rad damage mutation' thing? if any of you guys can come up with a better idea on how to separate the pools so coders just dont start adding things to both pools then eventually merging the pools, be my guest.

also, does the janitor's biosuit already protect from radiation? if not, add that in so it's a tool the janitor can use to actually clean up the messes they're supposed to.

**Comment by ghost** [comment]
The janitors biosuit inherits the regular radiation protection of biosuits, so it's 20% off assuming they're wearing everything. Biosuits are more for as the name implies biological mishaps

Celt got us covered with that by making the suit cover hands and feet, good shit

**Comment by SkowronX** [comment]
In terms of visible grotesque mutation, clone damage utilizes a separate dmi for misshapen body parts. Could probably use a similar method for tumors and ugly growths.

**Comment by ghost** [comment]
I was actually going to work on radiation effects but of course
You should give a chance for nosebleeds that are only noticeable by examining yourself when you have a small dose of radiation
High levels of radiation could come with flesh rotting off, hallucinations, cancerous growths, etc.

**Comment by ghost** [comment]
updated with some suggestions and places to put them. Any qualms or more ideas to add?

**Comment by Shadowmech88** [comment]
I think radiation as a whole should be reworked so that all `atom`s have a `radioactivity` var which gets incremented when exposed to radiation, and which irradiates all nearby atoms while decrementing, with a range and intensity relative to the value of its `radioactivity`.

That way radiation would become more of a polluting, dangerous hazard that would require maneuvering, containment, and decontamination rather than just being "don't be near thing or it poisons you."

**Comment by ghost** [comment]
In that case, probably best to have a can_be_irradiated proc so you don't get radioactive glass or something silly like that

**Comment by Blithering** [comment]
Can I put forward vodka as the ghetto radaway for fun times? Also radiation contamination as @Shadowmech88 said needs to be a thing. 
Pockets of the station becoming highly radioactive death zones after a radstorm would be neat too, have an engineering Geiger counter that can detect the areas and decontamination foam, while joe spaceman has to rely on some form of jumpsuit patch that changes colour and gives a message to chat when a certain value of radiation has been absorbed to let them know when they've been standing in it.
Basically make radiation invisible, slow acting death spots rather than maintenance huddle then anti tox the braindead or ghetto geneticists.

**Comment by ghost** [comment]
rads
Rads are a helluva drug

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 1]
```diff
@@ -73,3 +78,193 @@
...
+							if(isvox(src))
+								new_species_name = "Skeletal Vox"
+							else
+								new_species_name = "Skellington"
+							var/datum/species/S = all_species[new_species_name]
+							victim.species = new S.type
+							update_body()
+			if(prob(1*major_rad_multiplier))
+				//Rad glow
+				to_chat(src, "<span class = 'blob'>You start glowing!</span>")
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: Have not added actual radglow yet. Will do so now.

**Comment by Kurfursten** [comment]
The kind of shitty thing about vomitting is it stuns you, so my only worry is we're replacing one "stunned inside range of radiation thing" for another "stunning inside range of radiation thing"

Other than that, looks cool. Will this change be essentially unnoticeable to people who take just a few rads here and there? e.g.: engineers

**Comment by ghost** [comment]
Ailments don't start for the first 100 ticks, besides radiations toxin damage that it already deals, so the vomiting won't occur until you've been in the glow for about 10 seconds

**Comment by ghost** [comment]
paranoia frontier eleven 2017-03-28 115712
Biggest problem right now is a high exposure to rads kills you because of pure toxin damage

Supermatter really doesn't have that much of a rad range either

**Comment by Kurfursten** [comment]
That much is clearly by design, since it has a much larger rad range when stimulated with a thrown object. Current supermatter radiation is definitely tuned for containment.

**Comment by ghost** [comment]
you're doing the wish granter's work my friend
cheeki breeki, iv damki

vodka

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 2]
```diff
@@ -73,3 +89,200 @@
...
+						sdisabilities |= BLIND
+					else if (E.damage >= E.min_bruised_damage && !(disabilities & NEARSIGHTED))
+						simple_message("<span class='warning'>You go blind!</span>","<span class='warning'>Somebody turns the lights off.</span>")
+						eye_blind = 5
+						eye_blurry = 5
+						disabilities |= NEARSIGHTED
+						spawn(100)
+						disabilities &= ~NEARSIGHTED
+		if(rad_tick > 500)
+			if(prob(2.5*major_rad_multiplier))
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: Do these things even hurt you at all? I don't see anything for it and although I'm no expert, I'm pretty sure necrosis would be bad for your health.

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 3]
```diff
@@ -73,3 +89,200 @@
...
+							update_body()
+			if(prob(1*major_rad_multiplier))
+				//Rad glow
+				if(!(species.flags & RAD_GLOW))
+					to_chat(src, "<span class = 'blob'>You start glowing!</span>")
+					species.flags |= RAD_GLOW
+		if(rad_tick > 600)
+			if(prob(0.01*extreme_rad_multiplier))
+				//Ghoulification
+				to_chat(src, "<span class = 'notice'>You feel strangely at peace.</span>")
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: Should probably put this message after the if, otherwise it doesn't really mean anything.

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm (hunk 2 above)
Comment (ghost) [reply code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: I would think losing a limb would hurt, a lot

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm (hunk 3 above)
Comment (ghost) [reply code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: good spot

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm (hunk 2 above)
Comment (ghost) [reply code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: I mean it doesn't look like you added any extra brute damage or pain for losing flesh, which doesn't really make sense to me. Limbs are fine though because pain and whatnot is coded into the limb drop proc, and you're losing the entire limb anyway so it can't be damaged.

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm (hunk 3 above)
Comment (Kurfursten) [reply code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: If you're at 600 rad ticks you are about to be at peace one way or another

**Comment by Megamaw** [comment]
so does the janitor's biosuit now protect against rad messes? b/c it's kind of silly to have a very dangerous mess that the person hired to clean it up well, can't clean up

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 4]
```diff
@@ -73,3 +89,200 @@
...
 					O.add_autopsy_data("Radiation Poisoning", damage)
+	else
+		rad_tick = max(rad_tick-1,0)
+
+	if(rad_tick)
+		/*
+		A list of events that are more likely to occur based on time exposed and amount of exposure
+
+		6 'gates'
+		100
```
Comment (ComicIronic) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm:108]: Defines for your magic numbers.

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 5]
```diff
@@ -73,3 +89,200 @@
...
+				var/datum/organ/internal/eyes/E = internal_organs_by_name["eyes"]
+				if(!E.robotic && (sdisabilities & BLIND || disabilities & NEARSIGHTED))
+					to_chat(src, "<span class = 'danger'>[pick("Your eyesight starts to fade!","Your eyes go cloudy!","Are you going blind?")]</span>")
+					E.damage += 2.5
+					eye_blurry = min(eye_blurry+1.5,50)
+					if (E.damage >= E.min_broken_damage && !(sdisabilities & BLIND))
+						simple_message("<span class='warning'>You go blind!</span>","<span class='warning'>Somebody turns the lights off.</span>")
+						sdisabilities |= BLIND
+					else if (E.damage >= E.min_bruised_damage && !(disabilities & NEARSIGHTED))
+						simple_message("<span class='warning'>You go blind!</span>","<span class='warning'>Somebody turns the lights off.</span>")
```
Comment (ComicIronic) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: >you go blind

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 6]
```diff
@@ -73,3 +89,200 @@
...
+				"You feel like you're about to die!", \
+				"You feel every part of your body screaming in agony!", \
+				"A low, rolling pain passes through your body!", \
+				"Your body feels as if it's falling apart!", \
+				"You feel extremely weak!", \
+				"A sharp, deep pain bathes every inch of your body!")]")
+			if(prob(10)*minor_rad_multiplier)
+				//Blindness
+				var/datum/organ/internal/eyes/E = internal_organs_by_name["eyes"]
+				if(!E.robotic && (sdisabilities & BLIND || disabilities & NEARSIGHTED))
```
Comment (ComicIronic) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: Looks like you forgot another `!`, unless rads only makes eyes worse for the already poorly-sighted?

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 7]
```diff
@@ -73,3 +89,200 @@
...
+					if (E.damage >= E.min_broken_damage && !(sdisabilities & BLIND))
+						simple_message("<span class='warning'>You go blind!</span>","<span class='warning'>Somebody turns the lights off.</span>")
+						sdisabilities |= BLIND
+					else if (E.damage >= E.min_bruised_damage && !(disabilities & NEARSIGHTED))
+						simple_message("<span class='warning'>You go blind!</span>","<span class='warning'>Somebody turns the lights off.</span>")
+						eye_blind = 5
+						eye_blurry = 5
+						disabilities |= NEARSIGHTED
+						spawn(100)
+						disabilities &= ~NEARSIGHTED
```
Comment (ComicIronic) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: I'm surprised this works, it should be indented after the spawn. I guess it is valid syntax.

**Review by ComicIronic — DISMISSED** [review]
Need fixin'

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm (hunk 5 above)
Comment (ghost) [reply code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: What's wrong with this?

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm (hunk 7 above)
Comment (ghost) [reply code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: Pulled right from welder blinding, it's an odd place there

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm (hunk 5 above)
Comment (ComicIronic) [reply code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: The message is the same as for going blind, and the user is not going blind, they're getting another disability. It's confusing.

Comment (ghost) [reply code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: Well heck, I pulled it right from welding code same as the iffy spawn

Comment (ComicIronic) [reply code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: If you're copy-pasting code, make this into a proc.

**Comment by ghost** [comment]
@Megamaw No radioactive mess right now. Same old sources of radiation.

Maybe in part two

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm (hunk 4 above)
Comment (ghost) [reply code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm:108]: Defines defined

**Comment by ghost** [comment]
Okay, if nobody has any qualms with how radiation effects the person, we move onto ghetto alternatives for treating radiation.

**Comment by SkowronX** [comment]
vodka of course, or at least a recipe with it.

**Comment by ghost** [comment]
cheeki breeki soon

**Comment by ghost** [comment]
No doubt things may get nuclear in proportion due to large-scale testing being able to find things solo-testing would not.

unless there's any other issues or things that could be added, I think I'm just about done

File: code/modules/mob/mob_defines.dm [hunk 8]
```diff
@@ -170,6 +170,7 @@
 
 	var/datum/dna/dna = null//Carbon
 	var/radiation = 0.0//Carbon
+	var/rad_tick = 0.0//Carbon
```
Comment (unid15) [inline code/modules/mob/mob_defines.dm:173]: NO make this a /mob/living or a /mob/living/carbon variable

**Review by unid15 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 9]
```diff
@@ -73,3 +97,201 @@
...
+							victim.droplimb(1)
+					if(3)
+						//Skeletonify a limb
+						var/list/datum/organ/external/candidates = list()
+						for (var/datum/organ/external/O in organs)
+							if(O.status & ORGAN_ROBOT)
+								continue
+							if(O.amputated)
+								continue
+							if(O.species == "Skeletal Vox" || O.species == "Skellington")
```
Comment (unid15) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: istype(O.species, /datum/species/skellington)

all skeletal species are derived from /datum/species/skellington

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 10]
```diff
@@ -73,3 +97,201 @@
...
+
+		if(rad_tick > RADDOSELIGHT)
+			if(prob(5*rad_multiplier))
+				//Vomit
+				vomit()
+			if(prob(5*minor_rad_multiplier))
+				//Nosebleed
+				if(prob(15))
+					to_chat(src, "<span class = 'danger'>Your nose starts bleeding!</span>")
+				drip(1)
```
Comment (unid15) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: Change the drip proc to return 1 on success (it's in code/modules/organs/blood.dm @ 171), then only display the message if the drip is successful

otherwise species with no blood will get nosebleed messages which is going to destroy immersions bigly

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 11]
```diff
@@ -73,3 +97,201 @@
...
+			if(prob(5*major_rad_multiplier))
+				//Hallucination
+				hallucination += rand(1,5)*minor_rad_multiplier
+		if(rad_tick > RADDOSEMINOR)
+			if(prob(2*major_rad_multiplier))
+				//Internal hemorrhaging
+				var/list/limbs_to_bleed = list()
+				for(var/datum/organ/external/E in organs)
+					if(E.is_robotic())
```
Comment (unid15) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: if(!E.is_organic())

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 12]
```diff
@@ -73,3 +97,201 @@
...
+			if(prob(5*rad_multiplier))
+				//Minor clone damage
+				adjustCloneLoss(rand(1,5))
+				to_chat(src, "<span class='warning'>[pick("You can feel your body becoming weak!</span>", \
+				"You feel like you're about to die!", \
+				"You feel every part of your body screaming in agony!", \
+				"A low, rolling pain passes through your body!", \
+				"Your body feels as if it's falling apart!", \
+				"You feel extremely weak!", \
+				"A sharp, deep pain bathes every inch of your body!")]")
```
Comment (unid15) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: \</span\> should be here

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 13]
```diff
@@ -73,3 +97,201 @@
...
+
+							if(!Tx.density)
+								var/obj/effect/decal/cleanable/blood/gibs/D = getFromPool(/obj/effect/decal/cleanable/blood/gibs, Tx)
+								D.New(Tx,i)
+
+					if(2)
+						//Drop a limb
+						var/list/datum/organ/external/candidates = list()
+						for (var/datum/organ/external/O in organs)
+							if(O.status & ORGAN_ROBOT)
```
Comment (unid15) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: if(!O.is_organic())

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 14]
```diff
@@ -73,3 +97,201 @@
...
+								continue
+							candidates.Add(O)
+						if (candidates.len)
+							var/datum/organ/external/victim = pick(candidates)
+							victim.droplimb(1)
+					if(3)
+						//Skeletonify a limb
+						var/list/datum/organ/external/candidates = list()
+						for (var/datum/organ/external/O in organs)
+							if(O.status & ORGAN_ROBOT)
```
Comment (unid15) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: if(!O.is_organic())

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 15]
```diff
@@ -73,3 +97,201 @@
...
+						if(candidates.len)
+							var/datum/organ/external/victim = pick(candidates)
+							to_chat(src, "<span class = 'warning'>The flesh is falling off of your [victim.display_name]!</span>")
+							var/new_species_name
+							if(isvox(src))
+								new_species_name = "Skeletal Vox"
+							else
+								new_species_name = "Skellington"
+							var/datum/species/S = all_species[new_species_name]
+							victim.species = new S.type
```
Comment (unid15) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: Might be worth it turning this whole thing into a datum/organ/external proc. Either proc/set_species() or better yet proc/skeletify()

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 16]
```diff
@@ -73,3 +97,201 @@
...
+				if(!E.robotic && !(sdisabilities & BLIND || disabilities & NEARSIGHTED))
+					to_chat(src, "<span class = 'danger'>[pick("Your eyesight starts to fade!","Your eyes go cloudy!","Are you going blind?")]</span>")
+					E.damage += 2.5
+					eye_blurry = min(eye_blurry+1.5,50)
+					if (E.damage >= E.min_broken_damage && !(sdisabilities & BLIND))
+						simple_message("<span class='warning'>You go blind!</span>","<span class='warning'>Somebody turns the lights off.</span>")
+						sdisabilities |= BLIND
+					else if (E.damage >= E.min_bruised_damage && !(disabilities & NEARSIGHTED))
+						simple_message("<span class='warning'>It becomes hard to see for some reason.</span>","<span class='warning'>Somebody turns the lights off.</span>")
+						eye_blind = 5
```
Comment (unid15) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: Shouldn't this be a += instead of =

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 17]
```diff
@@ -73,3 +97,201 @@
...
+						for(var/i = 1 to totalslabs)
+							var/obj/item/weapon/reagent_containers/food/snacks/meat/human/newmeat = new
+							newmeat.name = sourcename + " " + newmeat.name
+							newmeat.subjectname = sourcename
+							newmeat.subjectjob = sourcejob
+							newmeat.reagents.add_reagent(NUTRIMENT, sourcenutriment / totalslabs) //Thehehe. Fat guys go first
+							//src.occupant.reagents.trans_to(newmeat, round (sourcetotalreagents / totalslabs, 1)) // Transfer all the reagents from the
+							allmeat[i] = newmeat
+
+							var/obj/item/meatslab = allmeat[i]
```
Comment (unid15) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: - the allmeat list appears to be completely useless
- instead of creating a meatslab variable that holds the reference to the created meat, just use the already existing newmeat variable
- instead of forceMoving the new meat slab, do newmeat = new(get_turf(src))
- I believe getFromPool() already calls New()

In addition everything in this entire block of code is copypasted from either the gibber or the butcher knife (or maybe even both aaaaa) please make it a proc

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 18]
```diff
@@ -73,3 +97,201 @@
...
+				//Become uncloneable
+				if(!(M_NOCLONE in mutations))
+					to_chat(src, "<span class = 'blob'>You feel something twist and break.</span>")
+					mutations |= M_NOCLONE
+		if(rad_tick > RADDOSECRITICAL)
+			if(prob(2*major_rad_multiplier))
+				//Minor limb mutation
+				var/list/datum/organ/external/candidates = list()
+				for (var/datum/organ/external/O in organs)
+					if(!(O.status & ORGAN_MUTATED|ORGAN_ROBOT))
```
Comment (unid15) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: This may mutate amputated limbs because heh organ code

if(O.is_usable() && O.is_organic())

is_usable() checks if the organ is destroyed or mutated

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 19]
```diff
@@ -73,3 +97,201 @@
...
+				//Minor clone damage
+				adjustCloneLoss(rand(1,5))
+				to_chat(src, "<span class='warning'>[pick("You can feel your body becoming weak!</span>", \
+				"You feel like you're about to die!", \
+				"You feel every part of your body screaming in agony!", \
+				"A low, rolling pain passes through your body!", \
+				"Your body feels as if it's falling apart!", \
+				"You feel extremely weak!", \
+				"A sharp, deep pain bathes every inch of your body!")]")
+			if(prob(10)*minor_rad_multiplier)
```
Comment (unid15) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: prob(10*minor_rad_multiplier)

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 20]
```diff
@@ -73,3 +97,201 @@
...
+							newmeat.subjectname = sourcename
+							newmeat.subjectjob = sourcejob
+							newmeat.reagents.add_reagent(NUTRIMENT, sourcenutriment / totalslabs) //Thehehe. Fat guys go first
+							//src.occupant.reagents.trans_to(newmeat, round (sourcetotalreagents / totalslabs, 1)) // Transfer all the reagents from the
+							allmeat[i] = newmeat
+
+							var/obj/item/meatslab = allmeat[i]
+							var/turf/Tx = get_turf(src)
+							meatslab.forceMove(get_turf(src))
+							meatslab.throw_at(Tx, i, 3)
```
Comment (unid15) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: why would you throw meat at the turf that it's already on, and increase the throw's range with each spawned meat (that's how the i variable is used here)

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 21]
```diff
@@ -73,3 +97,201 @@
...
+			if(prob(5*rad_multiplier))
+				//Organ damage
+				var/list/organ_to_damage = list()
+				for(var/datum/organ/internal/I in internal_organs)
+					if(I.robotic)
+						continue
+					organ_to_damage.Add(I)
+				if(organ_to_damage.len)
+					var/datum/organ/internal/victim = pick(organ_to_damage)
+					victim.damage += rand(1,5)*rad_multiplier
```
Comment (unid15) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: victim.take_damage(amount, silent=0)

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm [hunk 22]
```diff
@@ -73,3 +97,201 @@
...
+				"A low, rolling pain passes through your body!", \
+				"Your body feels as if it's falling apart!", \
+				"You feel extremely weak!", \
+				"A sharp, deep pain bathes every inch of your body!")]")
+			if(prob(10)*minor_rad_multiplier)
+				//Blindness
+				var/datum/organ/internal/eyes/E = internal_organs_by_name["eyes"]
+				if(!E.robotic && !(sdisabilities & BLIND || disabilities & NEARSIGHTED))
+					to_chat(src, "<span class = 'danger'>[pick("Your eyesight starts to fade!","Your eyes go cloudy!","Are you going blind?")]</span>")
+					E.damage += 2.5
```
Comment (unid15) [inline code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: E.take_damage(2.5, TRUE)

File: code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm (hunk 17 above)
Comment (ghost) [reply code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: it's copypasted from the necrosis virus2 effect

Comment (unid15) [reply code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm]: This means that this copypasta is in three different places already

File: code/modules/mob/mob_defines.dm (hunk 8 above)
Comment (ghost) [reply code/modules/mob/mob_defines.dm:173]: figured it'd be best kept with radiation, but okay

Comment (ghost) [reply code/modules/mob/mob_defines.dm:173]: Oh yeah, I use it in rejuvenate() as that resets radiation to 0 along with rad_tick

**Review by unid15 — APPROVED** [review]

**Comment by ghost** [comment]
Now, the only question is 

/WHEN/

**Comment by BlueNexus** [comment]
BayRadiation when

**Comment by PJB3005** [comment]
@BlueNexus did you fix the performance?

**Comment by ghost** [comment]
Absolute madegg
